packages/app/src/components/commons/NetworkModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { Button, CircularProgress, Grid, Modal, Typography, styled } from "@mui/material"
+import React, { useRef, useState } from "react"
+import { palette, typography } from "../../theme"
+import { ViewContainer } from "./ViewContainer"
+import CloseIcon from "@mui/icons-material/Close"
+import { useWeb3React } from "@web3-react/core"
+import { useNotification } from "../../hooks/useNotification"
+
+type NetworkError = {
+  code: number
+  message: string
+  stack: string
+}
+
+const ModalContainer = styled(ViewContainer)({
+  position: "absolute",
+  top: "50%",
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+  borderRadius: 8,
+  width: 648,
+  background: palette.whites[1000],
+  padding: 24,
+})
+
+const NetworkModal: React.FC<any> = ({ open, handleClose }) => {
+  const ref = useRef(null)
+  const openNotification = useNotification()
+  const { library } = useWeb3React()
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const handleSwitch = async () => {
+    if (library) {
+      try {
+        setLoading(true)
+        await library.send("wallet_switchEthereumChain", [{ chainId: "0x1" }])
+        handleClose({}, "escapeKeyDown")
```

Comment:
Used as MUIs `onClose` ModalProp, but looks like it is actually a `() => void` (from component usage in `packages/app/src/components/views/publication/components/SettingSection.tsx`).